### ISSUE TITLE
[Idea] Callback with ParseException

### ISSUE THREAD

**Issue by ScanEat** [phillwiggins/flutter_parse_sdk#37]
Hi,

I think it could be a good idea if for maybe most of functions (signup, reset password, cloud codes functions, etc.), we can have a callback with firstly the object(s) and secondly the ParseException, like this example (from the official docs: https://docs.parseplatform.org/android/guide/#retrieving-objects) :

```
ParseQuery<ParseUser> query = ParseUser.getQuery();
query.whereEqualTo("gender", "female");
query.findInBackground(new FindCallback<ParseUser>() {
  public void done(List<ParseUser> objects, ParseException e) {
    if (e == null) {
        // The query was successful.
    } else {
        // Something went wrong.
    }
  }
});
```
And
```
ParseUser.requestPasswordResetInBackground("myemail@example.com", new RequestPasswordResetCallback() {
  public void done(ParseException e) {
    if (e == null) {
      // An email was successfully sent with reset instructions.
    } else {
      // Something went wrong. Look at the ParseException to see what's up.
    }
  }
});
```
With this, if we sign up a user and there is an error, we know which error is because for the moment, we can't know why it fails.

**Comment by phillwiggins** [comment]
Okay so this is definitely something we can do. I did originally take this approach but then removed it as it added a lot of boiler plate code... however, after writing my own app using this library, I do feel that it's something that is needed.

This will be a breaking change but will be added.

I'm not 100% when the ParseUser queries will be added, but a ParseResponse object can be the return type for a ParseUser request.

**Comment by phillwiggins** [comment]
You were correct in this idea and this has now been implemented. It's on the develop branch and will be released with 1.0.6
